### ISSUE TITLE
pgcat decreases TPS from unproxied posgresql 16 by 4x to 5x (docker-compose reproducer)

### ISSUE THREAD

**Issue by cpbotha** [postgresml/pgcat#752]
**Describe the bug**
Performance of pgcat unexpectedly low on x86 hardware, but not on Apple Silicon.

| Platform        | baseline 10 | pgcat 10 | pgcat 100 |
| --------------- | ----------- | -------- | --------- |
| m1max           | 154         | 624      | 577       |
| ryzen 5900x wsl | 133         | 19       | 19        |
| i7-8750H ubuntu | 88          | 20       | 20        |

I hope I'm doing something wrong. :)

**To Reproduce**
docker-compose reproducer: https://github.com/cpbotha/pgcat-docker-compose-demo

From that readme:

Start the cluster:

```shell
docker-compose up
```

Try with `pgbench`:

```shell
# https://www.postgresql.org/docs/current/pgbench.html
# init "postgres" database for pgbench
pgbench -p 6432 -h localhost -i -U postgres postgres
# baseline perf with 10 clients to the bare postgres
pgbench -p 5432 -h localhost -T 20 -C -c 10 -n -U postgres postgres
# 10 clients via pgcat
pgbench -p 6432 -h localhost -T 20 -C -c 10 -n -U postgres postgres
# 100 clients via pgcat
# bench with 100 clients for 20 seconds
# -C: new connection for each transaction
# -n: no vacuuming before running test
pgbench -p 6432 -h localhost -T 20 -C -c 100 -n -U postgres postgres
```

**Expected behavior**

Going via pgcat should not decrease TPS by 4x to 5x like it does on x86.

**Comment by saolof** [comment]
Not a maintainer here, but I'm wondering about whether you pulled the image or if you tried building it locally from the dockerfile? Differences between architecture may or may not be fixable by tweaking the release profile and adding compilation flags or enabling/disabling jemalloc

**Comment by levkk** [comment]
I suspect the pgcat docker image is out of date since our Github action that publishes it is currently broken. There was a fix for this issue released recently in #749.
